bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/channel/KNXChannel.java

```diff
@@ -123,23 +123,29 @@ public final Set<GroupAddress> getWriteAddresses() {
             String dpt = Objects.requireNonNullElse(entry.getValue().getDPT(), getDefaultDPT(entry.getKey()));
             Set<Class<? extends Type>> expectedTypeClasses = DPTUtil.getAllowedTypes(dpt);
             // find the first matching type that is assignable from the command
-            for (Class<? extends Type> expectedTypeClass : expectedTypeClasses) {
-                if (expectedTypeClass.equals(command.getClass())) {
-                    logger.trace("getCommandSpec command class matches expected type class");
-                    return new WriteSpecImpl(entry.getValue(), dpt, command);
-                } else if (command instanceof State state && State.class.isAssignableFrom(expectedTypeClass)) {
-                    if (state.as(expectedTypeClass.asSubclass(State.class)) != null) {
-                        logger.trace("getCommandSpec command class is a sub-class of the expected type class");
-                        Class<? extends State> expectedTypeAsStateClass = expectedTypeClass.asSubclass(State.class);
-                        State convertedState = state.as(expectedTypeAsStateClass);
-                        if (convertedState != null) {
-                            return new WriteSpecImpl(entry.getValue(), dpt, convertedState);
+            if (expectedTypeClasses.contains(command.getClass())) {
+                logger.trace(
+                        "getCommandSpec key '{}' has one of the expectedTypeClasses '{}', matching command '{}' and dpt '{}'",
+                        entry.getKey(), expectedTypeClasses, command, dpt);
+                return new WriteSpecImpl(entry.getValue(), dpt, command);
+            } else {
+                for (Class<? extends Type> expectedTypeClass : expectedTypeClasses) {
+                    if (command instanceof State state && State.class.isAssignableFrom(expectedTypeClass)) {
+                        if (state.as(expectedTypeClass.asSubclass(State.class)) != null) {
+                            logger.trace(
+                                    "getCommandSpec command class {} is a sub-class of the expected type class {} for key {}",
+                                    command.getClass(), expectedTypeClass, entry.getKey());
+                            Class<? extends State> expectedTypeAsStateClass = expectedTypeClass.asSubclass(State.class);
+                            State convertedState = state.as(expectedTypeAsStateClass);
+                            if (convertedState != null) {
+                                return new WriteSpecImpl(entry.getValue(), dpt, convertedState);
+                            }
                         }
                     }
                 }
             }
         }
-        logger.trace("getCommandSpec no Spec found!");
+        logger.error("getCommandSpec no Spec found!");
```

Comment:
Maybe include some more info here, without a DPT or command it does not help end users to locate the problem.